### ISSUE TITLE
Fix  Dockerfile to leave validator running as a service

### ISSUE THREAD

**Issue by antonroman** [sonata-nfv/tng-sdk-validation#27]
We need to fix the Dockerfile to leave the validator running as a service so that it can be tested from Jenkins

**Comment by antonroman** [comment]
We need to install `tng-sdk-project` (to use workspace) in the dockerfile to install and execute` tng-sdk-validation `package. 
The code from `tng-sdk-validation` is directly copied from the root directory:

```
ADD . /tng-sdk-validation
WORKDIR /tng-sdk-validation
```
However to get the `tng-sdk-project` code I would need to clone and install it first. The problem I'm facing is that I would need to add an authorized private key to be able to clone it, at least while the repository is private. I tried to do sth like this:
```
WORKDIR /opt
RUN git clone git@github.com:sonata-nfv/tng-sdk-project.git
WORKDIR tng-sdk-project
RUN pip install -r requirements.txt
RUN python setup.py install
```
But then I realized I need to have access rights and adding a private key to the docker image doesn't look a good practice to me. I was checking the rest of dockerfiles of the sdk repositories but they seem not to have dependencies on each other. 

@StefanUPB @mpeuster @felipevicens any piece of advice is welcome :-)

**Comment by StefanUPB** [comment]
As far as I know, the `sonata-nfv/tng-sdk-project` repo is public, so you should be able to clone it via https without having to use any key:

```
git clone https://github.com/sonata-nfv/tng-sdk-project.git
```

**Comment by antonroman** [comment]
Thanks Stefan!,
 you are right, my fault.

**Comment by felipevicens** [comment]
Hello @antonroman I think we could optimize the size of the container. Currently it is 253MB but being an alpine base image probably we could reduce it. First thing that I've identified is that the folder gui inside src/tngsdk/validation weight 142MB and the reason is because containes the node_modules. Is the node_modules something that we need in the repo? Maybe we could add it to .gitignore and install it when we built the gui container. The current option is to add in .dockerignore the folders that we don't need for the python process. I would say:
* /doc
* /pipeline
* src/tngsdk/validation/gui

In the Dockerfile we could optimize the creation doing it in 2 steps. One to install the dependencies like git and clone the repo tng-sdk-project and a second one to copy from 1st container the tng-sdk-project and install it. So in this second container you don't need git.
